### ISSUE TITLE
[Bug]: Combobox options list overflows the screen top under small viewports

### ISSUE THREAD

**Issue by RamyRafik** [microsoft/fluentui#27565]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (20) x64 Intel(R) Xeon(R) W-2255 CPU @ 3.70GHz
    Memory: 37.63 GB / 63.69 GB
  Browsers:
    Edge: Spartan (44.22621.1555.0), Chromium (111.0.1661.62)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://codesandbox.io/s/runtime-river-pg2kgy?file=/example.tsx

## Actual Behavior
Currently, the listbox (Options list) has a max-height of 80vh. This would work if the Combobox is at the top or bottom of the page, but doesn't work if it's in the middle.
Under small viewports (where the listbox can't open to the bottom or right), the listbox would overflow the screen top edge.
![image](https://user-images.githubusercontent.com/20221435/231896441-b89faab8-f1de-4e3c-a4d4-64a5bc2f1751.png)

## Expected Behavior
The listbox should be always clamped by the screen edges.

### Requested priority

High

### Products/sites affected

PowerApps

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by smhigley** [comment]
This was just fixed in #27521, and should be available now :)

**Comment by smhigley** [comment]
Ah sorry, I misread the issue -- for comboboxes at screens small enough that they can't fit to the side, above, or below, you're right that they will be clipped.

I think there are two possible approaches:
1. We let the popup overlap the target when it can't fit on any side
2. We query the relative position of the trigger element in the document, and set max-height based on that.

This might also apply to Menu and Popover, since I don't believe we do any height-limiting based on available space in v9 yet.
